src/format.js

```diff
@@ -0,0 +1,29 @@
+import { pipe, pipeCompatibleReduce } from './utility.js';
+
+//line Object access layer
+const getText = (lineObj) => lineObj.text ?? '';
+const getSubMenuLevel = (lineObj) => lineObj.submenuLevel;
+// deno-lint-ignore no-unused-vars
+const ignoreSubMenuLevel = ({ submenuLevel, ...lineObj }) => lineObj;
+// deno-lint-ignore no-unused-vars
+const ignoreText = ({ text, ...lineObj }) => lineObj;
+
+//line formatting layer
+const getSubMenuFormatting = (lineObj) => '--'.repeat(getSubMenuLevel(lineObj));
+const getFormattedText = (lineObj) => `${getText(lineObj)}`.trim();
+const formatOutputOptions = pipeCompatibleReduce((output, [option, value]) =>
+  output.concat(` | ${option}=${value}`)
+)('');
+
+const getFormattedOutputOptions = pipe(
+  ignoreText,
+  ignoreSubMenuLevel,
+  Object.entries,
+  formatOutputOptions,
+);
+
+//entry layer
+export const formatLine = (lineObj) =>
```

Comment:
The intent of this function is transform an object into a formatted string. This function is composed of 3 separate functions that break this work into parts. When reviewing please review if it is clear that the work is spilt into parts and that the intent of each of these parts is clear.